src/lib/comms/sol-socket-impl-riot.c

```diff
@@ -53,10 +53,10 @@ ipv6_udp_recvmsg(struct sol_socket_riot *s, void *buf, size_t len, struct sol_ne
 
     SOL_NULL_CHECK(pkt, -EAGAIN);
 
-    /* When more types of sockets (not just datagram) are accepted,
-     * remember to err if !buf && type != DGRAM */
-    if (!buf)
-        return pkt->size;
+    if (!(buf->flags & (SOL_BUFFER_FLAGS_FIXED_CAPACITY | SOL_BUFFER_FLAGS_MEMORY_NOT_OWNED))) {
```

Comment:
btw this check is cumbersome, add a macro in `sol-buffer.h` like `SOL_BUFFER_CAN_RESIZE()`